client/lib/sites-list/index.js

```diff
@@ -16,7 +22,15 @@ module.exports = function() {
 				case 'RECEIVE_DELETED_SITE':
 					_sites.removeSite( action.site );
 					break;
-
+				case InvitesActionTypes.INVITE_ACCEPTED:
+					let sites = clone( _sites.get() );
+					if ( sites.length === 0 ) {
+						action.invite.site.primary = true;
+					}
+					sites.push( action.invite.site );
```

Comment:
It seems like we keep having errors that are likely due to us trying to add our idea of a site object into the sites list. We should try to find some way to normalize our data into a site object. Perhaps we should try creating a proper site object using [lib/site](https://github.com/Automattic/wp-calypso/blob/master/client/lib/site/README.md)?